qoqo-qryd/src/tweezer_devices.rs

```diff
@@ -413,11 +431,26 @@ impl TweezerDeviceWrapper {
     ///     str: The serialized form of TweezerDevice.
     ///
     /// Raises:
-    ///     ValueError: Cannot serialize TweezerDevice to json.
+    ///     ValueError: Cannot serialize TweezerDevice to json or
+    ///         the device does not have valid QRyd gates available.
     fn to_json(&self) -> PyResult<String> {
-        let serialized = serde_json::to_string(&self.internal)
-            .map_err(|_| PyValueError::new_err("Cannot serialize TweezerDevice to json"))?;
-        Ok(serialized)
+        const SINGLE_GATES: [&str; 2] = ["PhaseShiftState1", "RotateXY"];
+        const DOUBLE_GATES: &str = "PhaseShiftedControlledPhase";
+        if self.internal.layout_register.values().any(|info| {
+            SINGLE_GATES
+                .iter()
+                .any(|&g| info.tweezer_single_qubit_gate_times.contains_key(g))
+                || info.tweezer_two_qubit_gate_times.contains_key(DOUBLE_GATES)
+        }) {
+            let serialized = serde_json::to_string(&self.internal)
+                .map_err(|_| PyValueError::new_err("Cannot serialize TweezerDevice to json"))?;
+            Ok(serialized)
+        } else {
+            Err(PyValueError::new_err(
+                "The device does not allow any valid gate in any layout. ".to_owned() +
```

Comment:
This first line of the error message seems confusing, I know what you mean because of the docstring, but I feel it would benefit from being reworded